### DISCUSSION TITLE
ion interaction and segregation

### DISCUSSION THREAD

**Discussion by zahidhasan83** [idaholab/moose#28460] [Q&A Modules: Porous Flow]
Dear MOOSE users,
I am trying to simulate the interaction of two different ion types and their segregation in two separate pores, whereby each pore will accommodate one ion type. Should I use the chemical reaction module for this sort of computation? I would appreciate your advice. If you can direct me to the right module or examples I can follow to set up such a simulation in MOOSE, I will be immensely grateful to you.

Thanks, and best regards,
Zahid

**Comment by GiudGiud** [comment]
Hello

Is the spatial part of this process diffusion driven or advection driven?

**Reply by GiudGiud** [reply]
the chemical reactions module will handle the reaction term well, and will have utilities to set up chains of reactions if needed. For two species, it should be easy enough to set up the equations manually with Reaction kernels though

**Reply by zahidhasan83** [reply]
@GiudGiud The spatial part of the process is diffusion-limited. Which reaction kernels do you mean? Will these reaction kernels consider the repulsion of chemical species and their segregation?

**Reply by zahidhasan83** [reply]
@GiudGiud can these reaction kernels assign repulsion between two positive or two negative ions?

**Reply by GiudGiud** [reply]
I don't think we can do that with reaction terms?
What does repulsion look like in the equations?

**Reply by zahidhasan83** [reply]
@GiudGiud the repulsion force is based on the Coulomb's law.
![repulsion force](https://github.com/user-attachments/assets/fb7c9941-1fa6-48bc-92a2-cad40bc932c8)
Two positive ions will diffuse through and exert repulsion force on each other. Any suggestion on solution method in MOOSE?

**Reply by GiudGiud** [reply]
@permcody do you know who may have worked on this?
@cticenhour same question, though I realize the EM module is not about ion repulsion

**Reply by GiudGiud** [reply]
in any case this is a fairly simply form for the kernel, I could see a new kernel to model this.
The only difficulty is that there is not 2 individual ions here, but rather a concentration of ions in every mesh cell, so the force probably needs to be distributed in some manner

**Reply by cticenhour** [reply]
@zahidhasan83 Without seeing the full set of partial differential equations (PDEs) you are trying to solve, I am unsure what to say here. I would guess that the repulsion force would impact the electromigration term of a Nernst–Planck style equation modeling an individual particle concentration.

@mangerij do you have thoughts there?

**Reply by zahidhasan83** [reply]
@GiudGiud the distribution of ion concentration looks similar as you have described. I am thinking if I can apply electrostatic force incorporated in the EM module, coupled with the matdiffusion kernel??
@cticenhour Yes, the Nernst-Planck style equation is a reasonable one. The PDE looks like:
![Nernst-Planck equation](https://github.com/user-attachments/assets/8ac13bab-f98e-4dc2-a4f1-14583fdcc5c9)
Since there is no flow velocity, the advection term is gone. Another issue, ions are in a dilute solution. Therefore, if we would get a gradient in electric potential I am not sure! But ions will experience a minimal force similar to Coulomb's. Can I use the electrostatic contact condition interface-kernel in MOOSE for that? I am not quite sure. A bit lost here. Thanks for your advise.

**Reply by cticenhour** [reply]
A simple electrostatic potential solve can be done using one of our family of Diffusion objects with a Reaction object (or multiple) to form what would be required from each species to form the charge density term on the right-hand-side. To simulate a simple jump in potential due to differing bulk material properties, the ElectrostaticContactCondition object may be sufficient for your needs. The tunable parameters for the contact conductance explained in the documentation were tuned to fit electric field assisted sintering (EFAS) experimental data, so you might need to get an external reference for data to find a better fit to use it in your model. This might involve running the electrostatic solve separately, with some "dummy" inputs for your charge density term until you find conductance behavior you're happy with.

Once you have potential being solved, you can then couple it in to the N-P equation you presented above. Term (3) (the potential term) might require a custom MOOSE object (inheriting from Diffusion, similar to our other Diffusion family objects) in order to take in (from the input file, or defined in source code) the properties you have laid out in this screenshot as well as the variable representing the potential. Term (1) can be represented via the TimeDerivative object, and Term (2) seems like a fairly simple MatDiffusion or FunctionDiffusion term, depending on how the Diffusivity is defined.
